src/Compilers/VisualBasic/Portable/Binding/Binder_Symbols.vb

```diff
@@ -516,10 +511,15 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
 
                 If forwardedType IsNot Nothing Then
-                    If forwardedType.IsErrorType Then
-                        Dim diagInfo As DiagnosticInfo = DirectCast(forwardedType, ErrorTypeSymbol).ErrorInfo
-                        If diagInfo.Code = ERRID.ERR_TypeFwdCycle2 Then
-                            encounteredCycle = True
+                    If Not reportedAnError And forwardedType.IsErrorType Then
```

Comment:
>Not reportedAnError [](start = 23, length = 19)

Consider simply passing null (Nothing) bag if we are not interested in diagnostics.